R/ci.R

```diff
@@ -1,82 +1,104 @@
-#' @aliases NULL
-NULL
-
-#' @section `use_travis()`:
-#' Adds a basic `.travis.yml` to the top-level directory of a package. This is a
-#' configuration file for the [Travis CI](https://travis-ci.org/) continuous
-#' integration service.
-#' @param browse Open a browser window to enable automatic builds for the
-#'   package.
+#' @details
+#' By default the CI-services "Travis" (Linux) and "Appveyor"
+#' (Windows) will be set up. Basic `.travis.yml` and `appveyor.yml` files are
+#' added to the top-level directory of a package.
+#'
+#' This function is aimed at supporting the most common use cases.
+#' Users who require more control are advised to manually call the individual
+#' functions.
+#'
+#' @name use_ci
+#' @importFrom travis travis_set_pat
+#' @param path `[string]`\cr
+#'   The path to the repo to prepare.
+#' @param quiet `[flag]`\cr
+#'   Less verbose output? Default: `FALSE`.
+#' @param services `[character]`\cr
+#'   CI services to add.
 #' @export
 #' @rdname ci
-use_travis <- function(browse = interactive()) {
-  check_uses_github()
-
-  use_template(
-    "travis.yml",
-    ".travis.yml",
-    ignore = TRUE
-  )
+use_ci <- function(path = ".", quiet = FALSE,
+                   services = c("Travis", "Appveyor")) {
+  #' @details
+  #' The following steps will be run:
+  withr::with_dir(path, {
+    #' 1. If necessary, create a GitHub repository via [use_github()]
+    travis:::use_github_interactive() #FIXME: Export function in travis
+    stopifnot(uses_github())
+
+    if ("Travis" %in% services) {
+      #' 1. Enable Travis via [travis_enable()]
+      travis_enable()
+
+      #' 1. Create a default `.travis.yml` file
+      #'    (overwrite after confirmation in interactive mode only)
+      travis:::use_travis_yml() #FIXME: Export function in travis
+    }
 
-  travis_activate(browse)
-  use_travis_badge()
+    repo_type <- travis:::detect_repo_type() #FIXME: Export function in travis
 
-  invisible(TRUE)
-}
+    if ("Appveyor" %in% services) {
+      #' 1. Create a default `appveyor.yml` file
+      #'    (depending on repo type, overwrite after confirmation
+      #'    in interactive mode only)
+      if (travis:::needs_appveyor(repo_type)) travis:::use_appveyor_yml() #FIXME: Export function in travis
+    }
 
-use_travis_badge <- function() {
-  check_uses_github()
+    #' 1. Create a default `tic.R` file depending on the repo type
+    #'    (package, website, bookdown, ...)
+    use_tic(repo_type)
 
-  url <- glue("https://travis-ci.org/{github_repo_spec()}")
-  img <- glue("{url}.svg?branch=master")
+    #' 1. Enable deployment (if necessary, depending on repo type)
+    #'    via [use_travis_deploy()]
+    if (travis:::needs_deploy(repo_type)) use_travis_deploy() #FIXME: Export function in travis
 
-  use_badge("Travis build status", url, img)
+    #' 1. Create a GitHub PAT and install it on Travis CI via [travis_set_pat()]
+    travis_set_pat()
+  })
 }
 
-travis_activate <- function(browse = interactive()) {
```

Comment:
Keep, forward to `travis_enable()` and deprecate?

Reply:
And hence integrate `travis_enable()` from Travis?